### PR TITLE
run-all script consider each Eigen version of a cpp tool as a standalone tool.

### PR DESCRIPTION
**Pull request by novikov-alexander**
- run-all script consider each Eigen version of a cpp tool as a standalone tool.
- All non-eigen versions of **tools currently don't use "Light" postfix**. It is made for unification.  
- ADOLC was always compiling as non-eigen. Fixed.

### CODE REVIEW

**Comment by tomjaguarpaw** [comment]
> run-all script consider each Eigen version of a cpp tool as a standalone tool.

This is a good idea.

I don't follow the specific details of this change but I approve of the motivation!

File: ADBench/run-all.ps1 [hunk 1]
```diff
@@ -155,232 +155,211 @@ $datadir = "$dir/data"
...
-
-		$cmd = ""
-		$cmdargs = @($dir_in, $dir_out, $fn, $script:nruns_f, $script:nruns_J, $script:time_limit)
-		if ($this.type -eq "bin") {
-			$cmd = "$script:bindir/tools/$($this.name)/Tools-$($this.name)-$objective.exe"
-		} elseif ($this.type -eq "cpp") {
-            #temporary solution until this script refactoring
-			$cmd = "$script:bindir/src/cpp/runner/CppRunner.exe"
+    [string]$name
+    [ToolType]$type
```
Comment (awf) [inline ADBench/run-all.ps1]: Great change, thanks.

File: ADBench/run-all.ps1 [hunk 2]
```diff
@@ -155,232 +155,211 @@ $datadir = "$dir/data"
...
+        <#
+        .SYNOPSIS
+        Create a new Tool object to be run
+
+        .EXAMPLE
+        [Tool]::new("Finite", "bin", [ObjectiveType] "GMM, BA, Hand, LSTM", 0, 0)
+        This will create a Tool:
+        - called "Finite"
+        - run from binary executables
+        - runs all five tests
```
Comment (awf) [inline ADBench/run-all.ps1]: Does this comment need fixing?  "all five"?

File: tools/ADOLC/CMakeLists.txt [hunk 3]
```diff
@@ -26,17 +26,29 @@ function(build_adolc EXE_NAME)
...
+	unset(ADOLC_PREV_OUTPUT_DIRECTORY)
+endfunction(build_adolc_eigen)
+
 build_adolc("GMM-FULL" "DO_GMM_FULL" "DO_CPP")
 build_adolc("GMM-SPLIT" "DO_GMM_SPLIT" "DO_CPP")
 
 build_adolc("BA" "DO_BA_BLOCK" "DO_CPP")
-# build_adolc("BA-Eigen" "DO_BA_BLOCK" "DO_EIGEN")
+# build_adolc_eigen("BA" "DO_BA_BLOCK")
 # TODO this doesn't work
```
Comment (awf) [inline tools/ADOLC/CMakeLists.txt]: I would leave ADOLC lower in the priority list for the moment, e.g. PyTorch fixes would come before this TODO.

File: ADBench/run-all.ps1 (hunk 2 above)
Comment (novikov-alexander) [reply ADBench/run-all.ps1]: Thank you

File: tools/ADOLC/CMakeLists.txt (hunk 3 above)
Comment (novikov-alexander) [reply tools/ADOLC/CMakeLists.txt]: My changes broke ADOLC running at all that's why I made it working

**Pull request merged**